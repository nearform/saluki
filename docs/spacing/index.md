# Spacing

---

## Configuration

Configurable via;

```
spacing
```

## Defaults

```js
{
  small: '1rem',
  medium: '2rem',
  large: '3rem'
}
```
