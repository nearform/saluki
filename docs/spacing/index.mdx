# Spacing

---

## Rules

```js
spacing.margin.none
spacing.margin.auto
spacing.margin.small
spacing.margin.medium
spacing.margin.large

spacing.margin.vertical.none
spacing.margin.vertical.auto
spacing.margin.vertical.small
spacing.margin.vertical.medium
spacing.margin.vertical.large

spacing.margin.horizontal.none
spacing.margin.horizontal.auto
spacing.margin.horizontal.small
spacing.margin.horizontal.medium
spacing.margin.horizontal.large

spacing.margin.top.none
spacing.margin.top.auto
spacing.margin.top.small
spacing.margin.top.medium
spacing.margin.top.large

spacing.margin.right.none
spacing.margin.right.auto
spacing.margin.right.small
spacing.margin.right.medium
spacing.margin.right.large

spacing.margin.bottom.none
spacing.margin.bottom.auto
spacing.margin.bottom.small
spacing.margin.bottom.medium
spacing.margin.bottom.large

spacing.margin.left.none
spacing.margin.left.auto
spacing.margin.left.small
spacing.margin.left.medium
spacing.margin.left.large

spacing.padding.none
spacing.padding.small
spacing.padding.medium
spacing.padding.large

spacing.padding.vertical.none
spacing.padding.vertical.small
spacing.padding.vertical.medium
spacing.padding.vertical.large

spacing.padding.horizontal.none
spacing.padding.horizontal.small
spacing.padding.horizontal.medium
spacing.padding.horizontal.large

spacing.padding.top.none
spacing.padding.top.small
spacing.padding.top.medium
spacing.padding.top.large

spacing.padding.right.none
spacing.padding.right.small
spacing.padding.right.medium
spacing.padding.right.large

spacing.padding.bottom.none
spacing.padding.bottom.small
spacing.padding.bottom.medium
spacing.padding.bottom.large

spacing.padding.left.none
spacing.padding.left.small
spacing.padding.left.medium
spacing.padding.left.large
```

## Configuration

Configurable via;

```
spacing
```

## Defaults

```js
{
  margin: {
    none: '0',
    auto: 'auto',
    small: '1rem',
    medium: '2rem',
    large: '4rem'
  },
  padding: {
    none: '0',
    small: '1rem',
    medium: '2rem',
    large: '4rem'
  }
}
```
