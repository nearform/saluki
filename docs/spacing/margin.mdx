import { Margin } from './'

# Margin

---

## Example

<Margin>Hello</Margin>

```js
const Text = styled.h1`
  ${spacing.margin.horizontal.large}
`
```

## Rules

```js
spacing.margin.none
spacing.margin.auto
spacing.margin.small
spacing.margin.medium
spacing.margin.large

spacing.margin.vertical.none
spacing.margin.vertical.auto
spacing.margin.vertical.small
spacing.margin.vertical.medium
spacing.margin.vertical.large

spacing.margin.horizontal.none
spacing.margin.horizontal.auto
spacing.margin.horizontal.small
spacing.margin.horizontal.medium
spacing.margin.horizontal.large

spacing.margin.top.none
spacing.margin.top.auto
spacing.margin.top.small
spacing.margin.top.medium
spacing.margin.top.large

spacing.margin.right.none
spacing.margin.right.auto
spacing.margin.right.small
spacing.margin.right.medium
spacing.margin.right.large

spacing.margin.bottom.none
spacing.margin.bottom.auto
spacing.margin.bottom.small
spacing.margin.bottom.medium
spacing.margin.bottom.large

spacing.margin.left.none
spacing.margin.left.auto
spacing.margin.left.small
spacing.margin.left.medium
spacing.margin.left.large
```

## Configuration

Configurable via;

```
spacing.margin
```

## Defaults

```js
{
  none: '0',
  auto: 'auto',
  small: '1rem',
  medium: '2rem',
  large: '4rem'
}
```
