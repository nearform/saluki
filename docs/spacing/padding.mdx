import { Padding } from './'

# Padding

---

## Example

<Padding>Hello</Padding>

```js
const Text = styled.h1`
  ${spacing.padding.vertical.large}
`
```

## Rules

```js
spacing.padding.none
spacing.padding.small
spacing.padding.medium
spacing.padding.large

spacing.padding.vertical.none
spacing.padding.vertical.small
spacing.padding.vertical.medium
spacing.padding.vertical.large

spacing.padding.horizontal.none
spacing.padding.horizontal.small
spacing.padding.horizontal.medium
spacing.padding.horizontal.large

spacing.padding.top.none
spacing.padding.top.small
spacing.padding.top.medium
spacing.padding.top.large

spacing.padding.right.none
spacing.padding.right.small
spacing.padding.right.medium
spacing.padding.right.large

spacing.padding.bottom.none
spacing.padding.bottom.small
spacing.padding.bottom.medium
spacing.padding.bottom.large

spacing.padding.left.none
spacing.padding.left.small
spacing.padding.left.medium
spacing.padding.left.large
```

## Configuration

Configurable via;

```
spacing.padding
```

## Defaults

```js
{
  none: '0',
  small: '1rem',
  medium: '2rem',
  large: '4rem'
}
```
