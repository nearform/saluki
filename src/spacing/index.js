export const defaultRules = {
  none: '0',
  auto: 'auto',
  small: '1rem',
  medium: '2rem',
  large: '4rem'
}

export function combineRules(
  defaultRules,
  customRules = { padding: {}, margin: {} }
) {
  return {
    padding: {
      ...defaultRules,
      ...customRules.padding,
      top: {
        ...defaultRules,
        ...customRules.padding
      },
      right: {
        ...defaultRules,
        ...customRules.padding
      },
      bottom: {
        ...defaultRules,
        ...customRules.padding
      },
      left: {
        ...defaultRules,
        ...customRules.padding
      },
      vertical: {
        ...defaultRules,
        ...customRules.padding
      },
      horizontal: {
        ...defaultRules,
        ...customRules.padding
      }
    },
    margin: {
      ...defaultRules,
      ...customRules.margin,
      top: {
        ...defaultRules,
        ...customRules.margin
      },
      right: {
        ...defaultRules,
        ...customRules.margin
      },
      bottom: {
        ...defaultRules,
        ...customRules.margin
      },
      left: {
        ...defaultRules,
        ...customRules.margin
      },
      vertical: {
        ...defaultRules,
        ...customRules.margin
      },
      horizontal: {
        ...defaultRules,
        ...customRules.margin
      }
    }
  }
}

function generateRules(rules, cssProp) {
  const newRules = {
    top: {},
    right: {},
    bottom: {},
    left: {},
    vertical: {},
    horizontal: {}
  }

  for (const [key, value] of Object.entries(rules)) {
    if (
      key === 'top' ||
      key === 'right' ||
      key === 'bottom' ||
      key === 'left'
    ) {
      for (const [k, v] of Object.entries(rules[key])) {
        newRules[key][k] = {
          [`${cssProp}-${key}`]: v
        }
      }
    } else if (key === 'vertical') {
      for (const [k, v] of Object.entries(rules[key])) {
        newRules[key][k] = {
          [`${cssProp}-top`]: v,
          [`${cssProp}-bottom`]: v
        }
      }
    } else if (key === 'horizontal') {
      for (const [k, v] of Object.entries(rules[key])) {
        newRules[key][k] = {
          [`${cssProp}-right`]: v,
          [`${cssProp}-left`]: v
        }
      }
    } else {
      newRules[key] = {
        [cssProp]: value
      }
    }
  }

  return newRules
}

export default function(spacing) {
  return {
    padding: generateRules(spacing.padding, 'padding'),
    margin: generateRules(spacing.margin, 'margin')
  }
}
