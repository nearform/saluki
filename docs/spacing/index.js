import styled from 'styled-components/macro'
import { spacing } from '../../src'

export const Margin = styled.h1`
  ${spacing.margin.horizontal.large}
`

export const Padding = styled.h1`
  ${spacing.padding.vertical.large}
`
