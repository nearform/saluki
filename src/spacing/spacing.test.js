import spacing from './index'

describe('spacing', () => {
  describe('when no custom config is provided', () => {
    it('should return the default config', () => {
      expect(spacing()).toMatchSnapshot()
    })
  })

  describe('when custom config is provided', () => {
    describe('when the custom config overrides a value', () => {
      it('should replace the default value', () => {
        expect(spacing({ padding: { small: '2rem' } })).toMatchSnapshot()
      })
    })

    describe('when the custom config provides a new value', () => {
      it('should add the value', () => {
        expect(spacing({ padding: { verysmall: '.5rem' } })).toMatchSnapshot()
      })
    })
  })
})
