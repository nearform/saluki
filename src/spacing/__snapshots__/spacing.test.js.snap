// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`spacing when custom config is provided when the custom config overrides a value should replace the default value 1`] = `
Object {
  "margin": Object {
    "auto": Object {
      "margin": "auto",
    },
    "bottom": Object {
      "auto": Object {
        "margin-bottom": "auto",
      },
      "large": Object {
        "margin-bottom": "4rem",
      },
      "medium": Object {
        "margin-bottom": "2rem",
      },
      "none": Object {
        "margin-bottom": "0",
      },
      "small": Object {
        "margin-bottom": "1rem",
      },
    },
    "horizontal": Object {
      "auto": Object {
        "margin-left": "auto",
        "margin-right": "auto",
      },
      "large": Object {
        "margin-left": "4rem",
        "margin-right": "4rem",
      },
      "medium": Object {
        "margin-left": "2rem",
        "margin-right": "2rem",
      },
      "none": Object {
        "margin-left": "0",
        "margin-right": "0",
      },
      "small": Object {
        "margin-left": "1rem",
        "margin-right": "1rem",
      },
    },
    "large": Object {
      "margin": "4rem",
    },
    "left": Object {
      "auto": Object {
        "margin-left": "auto",
      },
      "large": Object {
        "margin-left": "4rem",
      },
      "medium": Object {
        "margin-left": "2rem",
      },
      "none": Object {
        "margin-left": "0",
      },
      "small": Object {
        "margin-left": "1rem",
      },
    },
    "medium": Object {
      "margin": "2rem",
    },
    "none": Object {
      "margin": "0",
    },
    "right": Object {
      "auto": Object {
        "margin-right": "auto",
      },
      "large": Object {
        "margin-right": "4rem",
      },
      "medium": Object {
        "margin-right": "2rem",
      },
      "none": Object {
        "margin-right": "0",
      },
      "small": Object {
        "margin-right": "1rem",
      },
    },
    "small": Object {
      "margin": "1rem",
    },
    "top": Object {
      "auto": Object {
        "margin-top": "auto",
      },
      "large": Object {
        "margin-top": "4rem",
      },
      "medium": Object {
        "margin-top": "2rem",
      },
      "none": Object {
        "margin-top": "0",
      },
      "small": Object {
        "margin-top": "1rem",
      },
    },
    "vertical": Object {
      "auto": Object {
        "margin-bottom": "auto",
        "margin-top": "auto",
      },
      "large": Object {
        "margin-bottom": "4rem",
        "margin-top": "4rem",
      },
      "medium": Object {
        "margin-bottom": "2rem",
        "margin-top": "2rem",
      },
      "none": Object {
        "margin-bottom": "0",
        "margin-top": "0",
      },
      "small": Object {
        "margin-bottom": "1rem",
        "margin-top": "1rem",
      },
    },
  },
  "padding": Object {
    "auto": Object {
      "padding": "auto",
    },
    "bottom": Object {
      "auto": Object {
        "padding-bottom": "auto",
      },
      "large": Object {
        "padding-bottom": "4rem",
      },
      "medium": Object {
        "padding-bottom": "2rem",
      },
      "none": Object {
        "padding-bottom": "0",
      },
      "small": Object {
        "padding-bottom": "2rem",
      },
    },
    "horizontal": Object {
      "auto": Object {
        "padding-left": "auto",
        "padding-right": "auto",
      },
      "large": Object {
        "padding-left": "4rem",
        "padding-right": "4rem",
      },
      "medium": Object {
        "padding-left": "2rem",
        "padding-right": "2rem",
      },
      "none": Object {
        "padding-left": "0",
        "padding-right": "0",
      },
      "small": Object {
        "padding-left": "2rem",
        "padding-right": "2rem",
      },
    },
    "large": Object {
      "padding": "4rem",
    },
    "left": Object {
      "auto": Object {
        "padding-left": "auto",
      },
      "large": Object {
        "padding-left": "4rem",
      },
      "medium": Object {
        "padding-left": "2rem",
      },
      "none": Object {
        "padding-left": "0",
      },
      "small": Object {
        "padding-left": "2rem",
      },
    },
    "medium": Object {
      "padding": "2rem",
    },
    "none": Object {
      "padding": "0",
    },
    "right": Object {
      "auto": Object {
        "padding-right": "auto",
      },
      "large": Object {
        "padding-right": "4rem",
      },
      "medium": Object {
        "padding-right": "2rem",
      },
      "none": Object {
        "padding-right": "0",
      },
      "small": Object {
        "padding-right": "2rem",
      },
    },
    "small": Object {
      "padding": "2rem",
    },
    "top": Object {
      "auto": Object {
        "padding-top": "auto",
      },
      "large": Object {
        "padding-top": "4rem",
      },
      "medium": Object {
        "padding-top": "2rem",
      },
      "none": Object {
        "padding-top": "0",
      },
      "small": Object {
        "padding-top": "2rem",
      },
    },
    "vertical": Object {
      "auto": Object {
        "padding-bottom": "auto",
        "padding-top": "auto",
      },
      "large": Object {
        "padding-bottom": "4rem",
        "padding-top": "4rem",
      },
      "medium": Object {
        "padding-bottom": "2rem",
        "padding-top": "2rem",
      },
      "none": Object {
        "padding-bottom": "0",
        "padding-top": "0",
      },
      "small": Object {
        "padding-bottom": "2rem",
        "padding-top": "2rem",
      },
    },
  },
}
`;

exports[`spacing when custom config is provided when the custom config provides a new value should add the value 1`] = `
Object {
  "margin": Object {
    "auto": Object {
      "margin": "auto",
    },
    "bottom": Object {
      "auto": Object {
        "margin-bottom": "auto",
      },
      "large": Object {
        "margin-bottom": "4rem",
      },
      "medium": Object {
        "margin-bottom": "2rem",
      },
      "none": Object {
        "margin-bottom": "0",
      },
      "small": Object {
        "margin-bottom": "1rem",
      },
    },
    "horizontal": Object {
      "auto": Object {
        "margin-left": "auto",
        "margin-right": "auto",
      },
      "large": Object {
        "margin-left": "4rem",
        "margin-right": "4rem",
      },
      "medium": Object {
        "margin-left": "2rem",
        "margin-right": "2rem",
      },
      "none": Object {
        "margin-left": "0",
        "margin-right": "0",
      },
      "small": Object {
        "margin-left": "1rem",
        "margin-right": "1rem",
      },
    },
    "large": Object {
      "margin": "4rem",
    },
    "left": Object {
      "auto": Object {
        "margin-left": "auto",
      },
      "large": Object {
        "margin-left": "4rem",
      },
      "medium": Object {
        "margin-left": "2rem",
      },
      "none": Object {
        "margin-left": "0",
      },
      "small": Object {
        "margin-left": "1rem",
      },
    },
    "medium": Object {
      "margin": "2rem",
    },
    "none": Object {
      "margin": "0",
    },
    "right": Object {
      "auto": Object {
        "margin-right": "auto",
      },
      "large": Object {
        "margin-right": "4rem",
      },
      "medium": Object {
        "margin-right": "2rem",
      },
      "none": Object {
        "margin-right": "0",
      },
      "small": Object {
        "margin-right": "1rem",
      },
    },
    "small": Object {
      "margin": "1rem",
    },
    "top": Object {
      "auto": Object {
        "margin-top": "auto",
      },
      "large": Object {
        "margin-top": "4rem",
      },
      "medium": Object {
        "margin-top": "2rem",
      },
      "none": Object {
        "margin-top": "0",
      },
      "small": Object {
        "margin-top": "1rem",
      },
    },
    "vertical": Object {
      "auto": Object {
        "margin-bottom": "auto",
        "margin-top": "auto",
      },
      "large": Object {
        "margin-bottom": "4rem",
        "margin-top": "4rem",
      },
      "medium": Object {
        "margin-bottom": "2rem",
        "margin-top": "2rem",
      },
      "none": Object {
        "margin-bottom": "0",
        "margin-top": "0",
      },
      "small": Object {
        "margin-bottom": "1rem",
        "margin-top": "1rem",
      },
    },
  },
  "padding": Object {
    "auto": Object {
      "padding": "auto",
    },
    "bottom": Object {
      "auto": Object {
        "padding-bottom": "auto",
      },
      "large": Object {
        "padding-bottom": "4rem",
      },
      "medium": Object {
        "padding-bottom": "2rem",
      },
      "none": Object {
        "padding-bottom": "0",
      },
      "small": Object {
        "padding-bottom": "1rem",
      },
      "verysmall": Object {
        "padding-bottom": ".5rem",
      },
    },
    "horizontal": Object {
      "auto": Object {
        "padding-left": "auto",
        "padding-right": "auto",
      },
      "large": Object {
        "padding-left": "4rem",
        "padding-right": "4rem",
      },
      "medium": Object {
        "padding-left": "2rem",
        "padding-right": "2rem",
      },
      "none": Object {
        "padding-left": "0",
        "padding-right": "0",
      },
      "small": Object {
        "padding-left": "1rem",
        "padding-right": "1rem",
      },
      "verysmall": Object {
        "padding-left": ".5rem",
        "padding-right": ".5rem",
      },
    },
    "large": Object {
      "padding": "4rem",
    },
    "left": Object {
      "auto": Object {
        "padding-left": "auto",
      },
      "large": Object {
        "padding-left": "4rem",
      },
      "medium": Object {
        "padding-left": "2rem",
      },
      "none": Object {
        "padding-left": "0",
      },
      "small": Object {
        "padding-left": "1rem",
      },
      "verysmall": Object {
        "padding-left": ".5rem",
      },
    },
    "medium": Object {
      "padding": "2rem",
    },
    "none": Object {
      "padding": "0",
    },
    "right": Object {
      "auto": Object {
        "padding-right": "auto",
      },
      "large": Object {
        "padding-right": "4rem",
      },
      "medium": Object {
        "padding-right": "2rem",
      },
      "none": Object {
        "padding-right": "0",
      },
      "small": Object {
        "padding-right": "1rem",
      },
      "verysmall": Object {
        "padding-right": ".5rem",
      },
    },
    "small": Object {
      "padding": "1rem",
    },
    "top": Object {
      "auto": Object {
        "padding-top": "auto",
      },
      "large": Object {
        "padding-top": "4rem",
      },
      "medium": Object {
        "padding-top": "2rem",
      },
      "none": Object {
        "padding-top": "0",
      },
      "small": Object {
        "padding-top": "1rem",
      },
      "verysmall": Object {
        "padding-top": ".5rem",
      },
    },
    "vertical": Object {
      "auto": Object {
        "padding-bottom": "auto",
        "padding-top": "auto",
      },
      "large": Object {
        "padding-bottom": "4rem",
        "padding-top": "4rem",
      },
      "medium": Object {
        "padding-bottom": "2rem",
        "padding-top": "2rem",
      },
      "none": Object {
        "padding-bottom": "0",
        "padding-top": "0",
      },
      "small": Object {
        "padding-bottom": "1rem",
        "padding-top": "1rem",
      },
      "verysmall": Object {
        "padding-bottom": ".5rem",
        "padding-top": ".5rem",
      },
    },
    "verysmall": Object {
      "padding": ".5rem",
    },
  },
}
`;

exports[`spacing when no custom config is provided should return the default config 1`] = `
Object {
  "margin": Object {
    "auto": Object {
      "margin": "auto",
    },
    "bottom": Object {
      "auto": Object {
        "margin-bottom": "auto",
      },
      "large": Object {
        "margin-bottom": "4rem",
      },
      "medium": Object {
        "margin-bottom": "2rem",
      },
      "none": Object {
        "margin-bottom": "0",
      },
      "small": Object {
        "margin-bottom": "1rem",
      },
    },
    "horizontal": Object {
      "auto": Object {
        "margin-left": "auto",
        "margin-right": "auto",
      },
      "large": Object {
        "margin-left": "4rem",
        "margin-right": "4rem",
      },
      "medium": Object {
        "margin-left": "2rem",
        "margin-right": "2rem",
      },
      "none": Object {
        "margin-left": "0",
        "margin-right": "0",
      },
      "small": Object {
        "margin-left": "1rem",
        "margin-right": "1rem",
      },
    },
    "large": Object {
      "margin": "4rem",
    },
    "left": Object {
      "auto": Object {
        "margin-left": "auto",
      },
      "large": Object {
        "margin-left": "4rem",
      },
      "medium": Object {
        "margin-left": "2rem",
      },
      "none": Object {
        "margin-left": "0",
      },
      "small": Object {
        "margin-left": "1rem",
      },
    },
    "medium": Object {
      "margin": "2rem",
    },
    "none": Object {
      "margin": "0",
    },
    "right": Object {
      "auto": Object {
        "margin-right": "auto",
      },
      "large": Object {
        "margin-right": "4rem",
      },
      "medium": Object {
        "margin-right": "2rem",
      },
      "none": Object {
        "margin-right": "0",
      },
      "small": Object {
        "margin-right": "1rem",
      },
    },
    "small": Object {
      "margin": "1rem",
    },
    "top": Object {
      "auto": Object {
        "margin-top": "auto",
      },
      "large": Object {
        "margin-top": "4rem",
      },
      "medium": Object {
        "margin-top": "2rem",
      },
      "none": Object {
        "margin-top": "0",
      },
      "small": Object {
        "margin-top": "1rem",
      },
    },
    "vertical": Object {
      "auto": Object {
        "margin-bottom": "auto",
        "margin-top": "auto",
      },
      "large": Object {
        "margin-bottom": "4rem",
        "margin-top": "4rem",
      },
      "medium": Object {
        "margin-bottom": "2rem",
        "margin-top": "2rem",
      },
      "none": Object {
        "margin-bottom": "0",
        "margin-top": "0",
      },
      "small": Object {
        "margin-bottom": "1rem",
        "margin-top": "1rem",
      },
    },
  },
  "padding": Object {
    "auto": Object {
      "padding": "auto",
    },
    "bottom": Object {
      "auto": Object {
        "padding-bottom": "auto",
      },
      "large": Object {
        "padding-bottom": "4rem",
      },
      "medium": Object {
        "padding-bottom": "2rem",
      },
      "none": Object {
        "padding-bottom": "0",
      },
      "small": Object {
        "padding-bottom": "1rem",
      },
    },
    "horizontal": Object {
      "auto": Object {
        "padding-left": "auto",
        "padding-right": "auto",
      },
      "large": Object {
        "padding-left": "4rem",
        "padding-right": "4rem",
      },
      "medium": Object {
        "padding-left": "2rem",
        "padding-right": "2rem",
      },
      "none": Object {
        "padding-left": "0",
        "padding-right": "0",
      },
      "small": Object {
        "padding-left": "1rem",
        "padding-right": "1rem",
      },
    },
    "large": Object {
      "padding": "4rem",
    },
    "left": Object {
      "auto": Object {
        "padding-left": "auto",
      },
      "large": Object {
        "padding-left": "4rem",
      },
      "medium": Object {
        "padding-left": "2rem",
      },
      "none": Object {
        "padding-left": "0",
      },
      "small": Object {
        "padding-left": "1rem",
      },
    },
    "medium": Object {
      "padding": "2rem",
    },
    "none": Object {
      "padding": "0",
    },
    "right": Object {
      "auto": Object {
        "padding-right": "auto",
      },
      "large": Object {
        "padding-right": "4rem",
      },
      "medium": Object {
        "padding-right": "2rem",
      },
      "none": Object {
        "padding-right": "0",
      },
      "small": Object {
        "padding-right": "1rem",
      },
    },
    "small": Object {
      "padding": "1rem",
    },
    "top": Object {
      "auto": Object {
        "padding-top": "auto",
      },
      "large": Object {
        "padding-top": "4rem",
      },
      "medium": Object {
        "padding-top": "2rem",
      },
      "none": Object {
        "padding-top": "0",
      },
      "small": Object {
        "padding-top": "1rem",
      },
    },
    "vertical": Object {
      "auto": Object {
        "padding-bottom": "auto",
        "padding-top": "auto",
      },
      "large": Object {
        "padding-bottom": "4rem",
        "padding-top": "4rem",
      },
      "medium": Object {
        "padding-bottom": "2rem",
        "padding-top": "2rem",
      },
      "none": Object {
        "padding-bottom": "0",
        "padding-top": "0",
      },
      "small": Object {
        "padding-bottom": "1rem",
        "padding-top": "1rem",
      },
    },
  },
}
`;
